PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java

```diff
@@ -52,6 +54,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         ButterKnife.bind(this, v);
         mListView = (ListView) v.findViewById(android.R.id.list);
         initList();
+        frameLayout.setFocusableInTouchMode(true);
+        frameLayout.requestFocus();
+        frameLayout.setOnKeyListener(new View.OnKeyListener() {
+            @Override
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_BACK && event.getAction() == KeyEvent.ACTION_DOWN && fragment != null) {
+                    FragmentManager fragmentManager = mActivity.getSupportFragmentManager();
+                    fragmentManager.beginTransaction().remove(fragment).commit();
+                    fragment = null;
+                    return true;
+                }
+                return false;
+            }
+        });
```

Comment:
This should definitely not be in the `MainActivity onCreateView`, far too much complexity that should be refactored.